### ISSUE TITLE
checkpoint_e10.pth

### ISSUE THREAD

**Issue by chenbolinstudent** [STVIR/pysot#80]
why after checkpoint_e10.pth, it load model and train?

**Comment by lb1100** [comment]
There are two stages in SiamRPN++ training. The first 10 epochs only trains head(RPN/MASK). The other 10 epochs fine tunes backbone. So there are some difference in optimizer.

**Comment by chenbolinstudent** [comment]
so on testing,  only focus on 11 to 20 ,not 1 to 10 ?

**Comment by lb1100** [comment]
We usually test 5~20, since first 5 epochs can hardly track objects correctly. Fine-tune sometimes won't bring performance gain, so we test some epochs before fine-tuning.

**Comment by chenbolinstudent** [comment]
when fine-tuning,which dataset do  you use?
how do do  fine-tuning?

**Comment by chenbolinstudent** [comment]
when fine-tuning,which dataset do you use?
how do you fine-tuning in detail?

**Comment by lb1100** [comment]
All of our experiments are using the same datasets, Youtube+VID+cooc+imageNetDet. 
https://github.com/STVIR/pysot/blob/master/experiments/siamrpn_alex_dwxcorr_16gpu/config.yaml#L8
That is an example of finetuning, where you can define when and what to fine-tune.

**Comment by MaxLin86** [comment]
> There are two stages in SiamRPN++ training. The first 10 epochs only trains head(RPN/MASK). The other 10 epochs fine tunes backbone. So there are some difference in optimizer.

I trained a model which backbone is mobilev2 without pretrained setting in BACKBONE CONFIG.
When I test checkpoint after 6th epoch, the result was really poor. The boxs on every frames were all in bottom-right corner for bag.avi.
If it is necessary to add "backbone pretrained" setting in BACKBONE CONFIG?

**Comment by chenbolinstudent** [comment]
There are two stages in SiamRPN++ training. The first 10 epochs only trains head(RPN/MASK). 
when only trains head(RPN) what is the dataset?
when only trains head(MASK) what is the dataset? how to train mask?

adn the other 10 epochs fine tunes backbone.

**Comment by chenbolinstudent** [comment]
There are two stages in SiamRPN++ training. The first 10 epochs only trains head(RPN/MASK).
when only trains head(RPN) what is the dataset?
when only trains head(MASK) what is the dataset? how to train mask?

**Comment by lb1100** [comment]
> I trained a model which backbone is mobilev2 without pretrained setting in BACKBONE CONFIG.
When I test checkpoint after 6th epoch, the result was really poor. The boxs on every frames were all in bottom-right corner when test bag.avi.
> If it is necessary to add "backbone pretrained" setting in BACKBONE CONFIG?
SiamRPN requires a ImageNet pretrained weight. Otherwise performance will be much worse.

**Comment by lb1100** [comment]
> There are two stages in SiamRPN++ training. The first 10 epochs only trains head(RPN/MASK).
when only trains head(RPN) what is the dataset?
when only trains head(MASK) what is the dataset? how to train mask?

As stated before, all SiamRPN++ experiments are using the same datasets, Youtube+VID+coco+imageNetDet for both stages.

As for SiamMask, currently pysot do not support training yet.

**Comment by chenbolinstudent** [comment]
If I want to continue train at checkpoint_e9.pth checkpoint_e10.pth checkpoint_e11.pth 

what should I change in code？

could you give the detail?

**Comment by lb1100** [comment]
In config.yaml.
```
...
TRAIN:
    START_EPOCH: 10
    RESUME: snapshot/checkpoint_e10.pth
...
```

**Comment by chenbolinstudent** [comment]
but if I load checkpoint_e10.pth  ,it do not go into 

        if epoch != idx // num_per_epoch + start_epoch:
            epoch = idx // num_per_epoch + start_epoch
...
            if cfg.BACKBONE.TRAIN_EPOCH == epoch:
                logger.info('start training backbone.')
                optimizer, lr_scheduler = build_opt_lr(model.module, epoch)
                logger.info("model\n{}".format(describe(model.module)))

when I train from 0 , can go into it after checkpoint_e10.pth.

**Comment by chenbolinstudent** [comment]
We usually test 5~20, since first 5 epochs can hardly track objects correctly. Fine-tune sometimes won't bring performance gain, so we test some epochs before fine-tuning.

why  Fine-tune sometimes won't bring performance gain ？

**Comment by lb1100** [comment]
The most gain from fine-tune is that regression becomes better after fine-tune. However, training data only contains few categories compares to ImageNet. The robustness will be a little worse after fine-tuning.

**Comment by chenbolinstudent** [comment]
If I change the model , and want to continue train base on the  dwxcorr/model.pth
should I trains head(RPN/MASK) ,
should I  fine tunes backbone ?

**Comment by chenbolinstudent** [comment]
when training 
5 17:14:03,076-rk0-train.py#261] Epoch: [1][80/7142] lr: 0.001000
60 / 142840 [0%]

how to smaller the 7142 and 142840 ?

**Comment by chenbolinstudent** [comment]
if I want to 2 epochs only trains head(RPN/MASK). and 2 epochs fine tunes backbone.
how do I set ?

**Comment by chenbolinstudent** [comment]
in the network ,what is the relations of the two stages in SiamRPN++ in detail?
@lb1100

**Comment by haishibei** [comment]
> > There are two stages in SiamRPN++ training. The first 10 epochs only trains head(RPN/MASK). The other 10 epochs fine tunes backbone. So there are some difference in optimizer.
> 
> I trained a model which backbone is mobilev2 without pretrained setting in BACKBONE CONFIG.
> When I test checkpoint after 6th epoch, the result was really poor. The boxs on every frames were all in bottom-right corner when test bag.avi.
> If it is necessary to add "backbone pretrained" setting in BACKBONE CONFIG?

hi, have you pretrained the backbone of mobilenet_v2 on your own?
